vm/systemSmartContracts/staking_test.go

```diff
@@ -1 +1,448 @@
+		StakeValue:    nil,
+	}
+
+	stakedRegistrationData := stakingData{
+		StartNonce:    0,
+		Staked:        true,
+		UnStakedNonce: 0,
+		BlsPubKey:     nil,
+		StakeValue:    nil,
+	}
+
+	stakeValue := big.NewInt(0)
+	eei, _ := NewVMContext(&mock.BlockChainHookStub{}, &mock.CryptoHookStub{})
+	eei.SetSCAddress([]byte("addr"))
+
+	stakingSmartContract, _ := NewStakingSmartContract(stakeValue, eei)
+	arguments := CreateVmContractCallInput()
+	arguments.Function = "unStake"
+	arguments.Arguments = []*big.Int{big.NewInt(100), big.NewInt(200)}
+	arguments.Header = &vmcommon.SCCallHeader{Number: big.NewInt(100)}
+	marshalizedExpectedRegData, _ := json.Marshal(&stakedRegistrationData)
+	stakingSmartContract.eei.SetStorage(arguments.CallerAddr, marshalizedExpectedRegData)
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.Ok, retCode)
+
+	var registrationData stakingData
+	data := stakingSmartContract.eei.GetStorage(arguments.CallerAddr)
+	err := json.Unmarshal(data, &registrationData)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedRegistrationData, registrationData)
+}
+
+func TestStakingSC_ExecuteFinalizeUnStakeOwnerAddrNotOkShouldErr(t *testing.T) {
+	t.Parallel()
+
+	stakeValue := big.NewInt(0)
+	eei := &mock.SystemEIStub{}
+	stakingSmartContract, _ := NewStakingSmartContract(stakeValue, eei)
+	arguments := CreateVmContractCallInput()
+	arguments.Function = "finalizeUnStake"
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+}
+
+func TestStakingSC_ExecuteFinalizeUnStakeUnmarshalErr(t *testing.T) {
+	t.Parallel()
+
+	stakeValue := big.NewInt(0)
+	eei := &mock.SystemEIStub{}
+	eei.GetStorageCalled = func(key []byte) []byte {
+		return []byte("data")
+	}
+	stakingSmartContract, _ := NewStakingSmartContract(stakeValue, eei)
+	arguments := CreateVmContractCallInput()
+	arguments.CallerAddr = []byte("data")
+	arguments.Function = "finalizeUnStake"
+	arguments.Arguments = []*big.Int{big.NewInt(100), big.NewInt(200)}
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+}
+
+func TestStakingSC_ExecuteFinalizeUnStakeValidatorNotUnStakeShouldErr(t *testing.T) {
+	t.Parallel()
+
+	stakeValue := big.NewInt(0)
+	eei := &mock.SystemEIStub{}
+	eei.GetStorageCalled = func(key []byte) []byte {
+		switch {
+		case bytes.Equal(key, []byte(ownerKey)):
+			return []byte("data")
+		default:
+			registrationDataMarshalized, _ := json.Marshal(&stakingData{UnStakedNonce: 0})
+			return registrationDataMarshalized
+		}
+	}
+
+	stakingSmartContract, _ := NewStakingSmartContract(stakeValue, eei)
+	arguments := CreateVmContractCallInput()
+	arguments.CallerAddr = []byte("data")
+	arguments.Function = "finalizeUnStake"
+	arguments.Arguments = []*big.Int{big.NewInt(100)}
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+}
+
+func TestStakingSC_ExecuteFinalizeUnStake(t *testing.T) {
```

Comment:
Suggestion: create a test that performs both the `unstake` and `finalizeUnstake` operations in succession. This test simulates a prior `unstake`, then performs `finalizeUnstake`.